phpBB/includes/functions_mcp.php

```diff
@@ -51,8 +51,7 @@ function phpbb_module_warn_url($mode, $module_row)
 	{
 		global $forum_id, $post_id;
 
-		$url_extra = ($forum_id) ? "&amp;f=$forum_id" : '';
-		$url_extra .= ($post_id) ? "&amp;p=$post_id" : '';
+		$url_extra .= ($post_id) ? "&amp;p=$post_id" : (($forum_id) ? "&amp;f=$forum_id" : '');
```

Comment:
Better use conditional blocks instead of nested ternary operators to make it more readable.

Also, php [behaves weird](http://www.phpsadness.com/sad/30) when there are nested ternary operators, I wouldn't use them besides the parenthesis

Reply:
OK, will do.

Reply:
`$url_extra .= $post_id ? "&amp;p=$post_id" : ($forum_id ? "&amp;f=$forum_id" : '');` ?